pkg/repo/badgerdb/translation.go

```diff
@@ -20,27 +20,54 @@ func translationKey(serviceID uuid.UUID, language language.Tag) []byte {
 	return []byte(fmt.Sprintf("%s%s:%s", translationPrefix, serviceID, language))
 }
 
+func (r *Repo) Tx(ctx context.Context, fn func(repo.TranslationsRepo) error) error {
+	// start transaction
+	tx := r.db.NewTransaction(true)
+
+	defer func() {
+		if r := recover(); r != nil {
+			tx.Discard()
+		}
+	}()
+
+	if err := fn(&Repo{db: r.db, tx: tx}); err != nil {
+		tx.Discard()
+
+		return fmt.Errorf("repo: execute tx: %w", err)
+	}
+
+	if err := tx.Commit(); err != nil {
+		return fmt.Errorf("repo: commit tx: %w", err)
+	}
+
+	return nil
+}
```

Comment:
When `fn()` panics, no error is returned - false positive. 

Let's return an error when it panics.

PS. The code is self-descriptive, there's no need for a comment that we start a tx. :)


```suggestion
func (r *Repo) Tx(ctx context.Context, fn func(repo.TranslationsRepo) error) (err error) {
	tx := r.db.NewTransaction(true)

	defer func() {
		if r := recover(); r != nil {
			tx.Discard()
			err = fmt.Errorf("repo: tx panicked: %v", r)
		}
	}()

	if err = fn(&Repo{db: r.db, tx: tx}); err != nil {
		tx.Discard()

		return fmt.Errorf("repo: execute tx: %w", err)
	}

	if err = tx.Commit(); err != nil {
		return fmt.Errorf("repo: commit tx: %w", err)
	}

	return nil
}
```

Reply:
Let's pass ctx, too.

```go
func (r *Repo) Tx(ctx context.Context, fn func(ctx context.Context, repo.TranslationsRepo) error) (err error) {
```